components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Edge.Publisher/src/Engine/MonitoredItemMessageEncoder.cs

```diff
@@ -316,30 +322,43 @@ public class MonitoredItemMessageEncoder : IMessageEncoder {
             IEnumerable<DataSetMessageModel> messages, MessageEncoding encoding) {
             foreach (var message in messages) {
                 if (message.WriterGroup?.MessageType.GetValueOrDefault(MessageEncoding.Json) == encoding) {
+                    int notificationCounter = 0;
                     foreach (var notification in message.Notifications) {
```

Comment:
Not that important but for loop is faster then foreach. Since you need a counter anyways would be better to use a for loop